systemd/ceph-mon@.service.in

```diff
@@ -1,13 +1,14 @@
 [Unit]
 Description=Ceph cluster monitor daemon
-
+PartOf=ceph-mon.target
 # According to:
 #   http://www.freedesktop.org/wiki/Software/systemd/NetworkTarget
 # these can be removed once ceph-mon will dynamically change network
 # configuration.
 After=network-online.target local-fs.target time-sync.target
-Wants=network-online.target local-fs.target time-sync.target
-PartOf=ceph-mon.target
+Before=ceph-mon.target
+Wants=network-online.target local-fs.target time-sync.target ceph-mon.target
+RequiresMountsFor=/var/lib/ceph
```

Comment:
This path is dependent on how the Ceph `mon data` path is configured. Can't we just rely on the `local-fs.target` to mount everything beforehand here?

Reply:
Since we have `mon data = /var/lib/ceph` as default for `/etc/ceph/ceph.conf` unless expressly define and override, setting `RequiresMountsFor=/var/lib/ceph` could also be a reasonable default?

Similar idea, for advanced user that may possibly (well but rare) locate `mon data` into a remote mount point (!?), specifying `local-fs.target` only may not be enough?

In case for advanced user that going to override `mon data`, shall we also assume that they have complete idea and could update here accordingly?

Reply:
> In case for advanced user that going to override `mon data`, shall we also assume that they have complete idea and could update here accordingly?

I don't think it's practical to let user alter the systemd unit. Maybe its better to pass this via the `EnvironmentFile=` directive? Though I'm not sure this is available on the `[Unit]` section.

Reply:
Ok removed